src/streams.h

```diff
@@ -296,6 +296,16 @@ class CDataStream
         data.insert(data.end(), begin(), end());
         clear();
     }
+
+    /**
+     * XOR the contents of this stream with a certain key.
+     *
+     * @param[in] key    The key used to XOR the data in this stream.
+     */
+    void Xor(const std::string& key) {
```

Comment:
I think you misunderstand... we use a string consisting entirely of null-terminators (`\000`) as a default obfuscation key. Given https://coderpad.io/2GF6EKCD , the default obfuscation key would cause your code above to raise an assertion error. I went with an early return to avoid this.